src/tests/test_double_compare.cpp

```diff
@@ -0,0 +1,208 @@
+#include <boost/math/special_functions/next.hpp>
+#include <gtest/gtest.h>
+
+#include "types.h"
+
+namespace tests {
+
+class DoubleCompare : public ::testing::Test {
+private:
+    std::unique_ptr<std::byte[]> double_min_ptr;
+    std::unique_ptr<std::byte[]> double_max_ptr;
+    std::unique_ptr<std::byte[]> result_ptr;
+
+protected:
+    model::DoubleType double_type;
+    model::INumericType& double_type_ref = double_type;
+
+    std::byte* double_min;
+    std::byte* double_max;
+    std::byte* result;
+
+    void SetUp() override {
+        double_min_ptr.reset(double_type.MakeValue(std::numeric_limits<model::Double>().min()));
+        double_max_ptr.reset(double_type.MakeValue(std::numeric_limits<model::Double>().max()));
+        result_ptr.reset(double_type.MakeValue(0.0));
+        result = result_ptr.get();
+        double_min = double_min_ptr.get();
+        double_max = double_max_ptr.get();
+    }
+
+    void TestArithmetic(model::Double left, model::Double right, model::Double expected,
+                        std::function<void(std::byte*, std::byte*, std::byte*)> operation) {
```

Comment:
nit: In this situation, template is a better approach than `std::function` ([rationale](https://stackoverflow.com/questions/14677997/stdfunction-vs-template)).